anomalib/models/components/base/anomaly_module.py

```diff
@@ -36,30 +35,41 @@ class AnomalyModule(pl.LightningModule, ABC):
     Acts as a base class for all the Anomaly Modules in the library.
 
     Args:
-        params (Union[DictConfig, ListConfig]): Configuration
+        adaptive_threshold (Optional[bool], optional): Boolean to automatically choose adaptive threshold.
+            Defaults to None.
+        default_image_threshold (Optional[float], optional): Manual default image threshold.
+            Defaults to None.
+        default_pixel_threshold (Optional[float], optional): Manaul default pixel threshold.
+            Defaults to None.
     """
 
-    def __init__(self, params: Union[DictConfig, ListConfig]):
-
+    def __init__(
+        self,
+        adaptive_threshold: Optional[bool] = None,
+        default_image_threshold: Optional[float] = None,
+        default_pixel_threshold: Optional[float] = None,
+    ):
         super().__init__()
-        # Force the type for hparams so that it works with OmegaConfig style of accessing
-        self.hparams: Union[DictConfig, ListConfig]  # type: ignore
-        self.save_hyperparameters(params)
+        self.save_hyperparameters()
+        self.model: nn.Module
         self.loss: Tensor
         self.callbacks: List[Callback]
 
-        self.image_threshold = AdaptiveThreshold(self.hparams.model.threshold.image_default).cpu()
-        self.pixel_threshold = AdaptiveThreshold(self.hparams.model.threshold.pixel_default).cpu()
+        self.adaptive_threshold = False if adaptive_threshold is None else adaptive_threshold
+        self.default_image_threshold = 0.0 if default_image_threshold is None else default_image_threshold
+        self.default_pixel_threshold = 0.0 if default_pixel_threshold is None else default_pixel_threshold
+
+        self.image_threshold = AdaptiveThreshold(self.default_image_threshold).cpu()
+        self.pixel_threshold = AdaptiveThreshold(self.default_pixel_threshold).cpu()
```

Comment:
So these would be moved to the callback, and here in the base class we just create placeholders like we do for the MetricCollections below.

Reply:
would be good. With a quick try, it gave some issues when loading models. Unfortunately, I don't think I will enough time today to investigate what's wrong